owncloudApp/src/main/java/com/owncloud/android/presentation/files/filelist/MainFileListViewModel.kt

```diff
@@ -82,6 +83,10 @@ class MainFileListViewModel(
     private val sortTypeAndOrder = MutableStateFlow(Pair(SortType.SORT_TYPE_BY_NAME, SortOrder.SORT_ORDER_ASCENDING))
     val space: MutableStateFlow<OCSpace?> = MutableStateFlow(null)
 
+    private val _spaces: MutableStateFlow<List<OCSpace>> = MutableStateFlow(emptyList())
+    val spaces: StateFlow<List<OCSpace>>
+        get() = _spaces
+
```

Comment:
At the moment, we have an observable for the `currentFolderDisplayed`, another one for its `space`, and now we have an observable for every single space in the app, not only the ones from this account, to match each file with the space it belongs.

I think that we could introduce the space associated with each file within a wrapper as we already did with `OCFileWithFileSync`. It could contain also the space associated. There are already foreign keys, so the database is already prepared for this. If we introduce this, we could get rid of the space, and spaces observables since they would not be needed as we already had the space info for each file 🤔

Reply:
This is a bit complex and implies some changes throughout the code. I will create a new issue to address this 👍